torch/nn/modules/activation.py

```diff
@@ -741,19 +744,36 @@ def _reset_parameters(self):
     @weak_script_method
     def forward(self, query, key, value, key_padding_mask=None, incremental_state=None,
                 need_weights=True, static_kv=False, attn_mask=None):
-        """
-        Inputs of forward function
-            query: [target length, batch size, embed dim]
-            key: [sequence length, batch size, embed dim]
-            value: [sequence length, batch size, embed dim]
-            key_padding_mask: if True, mask padding based on batch size
-            incremental_state: if provided, previous time steps are cashed
-            need_weights: output attn_output_weights
-            static_kv: key and value are static
-
-        Outputs of forward function
-            attn_output: [target length, batch size, embed dim]
-            attn_output_weights: [batch size, target length, sequence length]
+        r"""
+    Args:
+        query, key, value: map a query and a set of key-value pairs to an output. 
+            See "Attention Is All You Need" for more details. 
+        key_padding_mask: if provided, padding elements can be excluded from by 
```

Comment:
Excluded from what?

Reply:
I.e. put the context of what it is excluded from in the documenation

Reply:
Padding elements can be excluded from the key by passing a binary ByteTensor

Reply:
Maybe a clearer meaning would be to say that the mask causes the attention to ignore padding elements.

Reply:
see the updates in the latest commit.